sssd_test_framework/roles/ipa.py

```diff
@@ -1597,3 +1633,89 @@ def __get_info(self, info: str | NFSExport | IPAAutomountMap | None) -> str | No
             return info.name
 
         return info
+
+
+class IPAPasswordPolicy(IPAObject, GenericPasswordPolicy):
+    """
+    Password policy management.
+    """
+
+    def __init__(self, role: IPA, name: str = "ipausers"):
+        """
+        :param role: IPA host object.
+        :type role: IPAHost
+        :param name: Name of target object, defaults to 'ipausers'.
+        :type name: str
+        """
+        super().__init__(role, name, command_group="pwpolicy")
+
+    def complexity(self, enable: bool) -> IPAPasswordPolicy:
+        """
+        Enable or disable password complexity.
+
+        :param enable: Enable or disable password complexity.
+        :type enable: bool
+        :return: IPAPasswordPolicy object.
+        :rtype: IPAPasswordPolicy
+        """
+        attrs: CLIBuilderArgs
+
+        if enable and self.get() is None:
+            attrs = {
+                "dictcheck": (self.cli.option.VALUE, "True"),
+                "usercheck": (self.cli.option.VALUE, "True"),
+                "minlength": (self.cli.option.VALUE, 8),
+                "minclasses": (self.cli.option.VALUE, 5),
+                "priority": (self.cli.option.VALUE, 1),
+            }
+            self._add(attrs)
+        else:
+            attrs = {
+                "dictcheck": (self.cli.option.VALUE, "False"),
+                "usercheck": (self.cli.option.VALUE, "False"),
+                "minlength": (self.cli.option.VALUE, 0),
+                "minclasses": (self.cli.option.VALUE, 0),
+                "priority": (self.cli.option.VALUE, 1),
+            }
+            self._modify(attrs)
```

Comment:
If `enable` is `false` and `self.get()` is `None`, you are calling `self._modify()` wheh you should call `self._add()`.

I think it would be better to prepare the attributes based on the value of `enable` and then call the right function based on whether `self.get()` is `None` or not.

Something like this:

```
if enable:
    attrs = {
        ...
    }
else:
    attrs = {
        ...
    }

if self.get() is None:
    self._add(attrs)
else:
    self._modify(attrs)
```